--- OperationalBase.cc.orig	2017-09-11 21:35:48.172823095 +0200
+++ OperationalBase.cc	2017-09-03 09:04:36.000000000 +0200
@@ -50,12 +50,17 @@
 void OperationalBase::handleMessageWhenDown(cMessage *message)
 {
     if (message->isSelfMessage())
-        throw cRuntimeError("Self message '%s' received when %s is down", message->getName(), getComponentType()->getName());
-    else if (simTime() == lastChange)
+        // following line is commented through errors when running out of power and mac use ack
+        // now we send a message instead of throwing a runtime error, dunno if it's a dump hack ;)
+        //throw cRuntimeError("Self message '%s' received when %s is down", message->getName(), getComponentType()->getName());
+        EV_WARN << "Self message " << message->getName() << " received when " << getComponentType()->getName() << " is down" << endl;
+    else if (simTime() == lastChange) {
         EV_WARN << getComponentType()->getName() << " is down, dropping '" << message->getName() << "' message\n";
-    else
+        delete message;
+    } else {
         throw cRuntimeError("Message '%s' received when %s is down", message->getName(), getComponentType()->getName());
-    delete message;
+        delete message;
+    }
 }
 
 bool OperationalBase::handleOperationStage(LifecycleOperation *operation, int stage, IDoneCallback *doneCallback)
